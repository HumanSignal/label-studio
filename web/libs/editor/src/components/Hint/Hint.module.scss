.main {
  font-size: 9px;
}
