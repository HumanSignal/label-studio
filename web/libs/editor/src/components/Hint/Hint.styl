.hint
  font-size 9px
