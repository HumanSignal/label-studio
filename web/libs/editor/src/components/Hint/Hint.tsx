import { CSSProperties, FC } from 'react';
import { Block } from '../../utils/bem';

import './Hint.styl';

interface HintProps {
  copy?: string;
  style?: CSSProperties;
  className?: string;
}

/**
 * Hint Component
 */
const Hint: FC<HintProps> = (props) => {
  return (
    <Block
      name="hint"
      tag="sup"
      className={props.className}
      data-copy={props.copy}
      style={props.style}
    >
      {props.children}
    </Block>
  );
};

export default Hint;
