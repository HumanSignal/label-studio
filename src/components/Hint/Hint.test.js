import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

import Hint from "./Hint";

describe("Hint", () => {
  it("Should render correctly", () => {
    const output = shallow(
      <Hint copy="test" style={{ background: "red" }} className="test">
        Test
      </Hint>,
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
