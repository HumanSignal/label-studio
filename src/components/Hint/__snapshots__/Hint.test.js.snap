// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hint Should render correctly 1`] = `
<sup
  className="main test"
  data-copy="test"
  style={
    Object {
      "background": "red",
    }
  }
>
  Test
</sup>
`;
